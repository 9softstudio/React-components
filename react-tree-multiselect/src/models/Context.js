import React from 'react';

export default React.createContext({
    rootNode: {},
    nodes: [],
    onToggle: () => { },
    onCheck: () => { }
});