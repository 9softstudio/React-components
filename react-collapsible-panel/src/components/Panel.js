import React from 'react';
import PropTypes from 'prop-types';
import { mergeClassName } from '../utils';
import Context from '../panel-context';

class Panel extends React.Component {
    constructor(props) {
        super(props);

        this.onToggleCollapse = this.onToggleCollapse.bind(this);
        this.state = {
            isExpanded: props.defaultExpanding,
            collapsible: props.collapsible
        };
    }

    onToggleCollapse() {
        if (this.props.collapsible) {
            this.setState({ isExpanded: !this.state.isExpanded });
        }
    }

    render() {
        return (<Context.Provider value={({ ...this.state, onToggleCollapse: this.onToggleCollapse })}>
            <div className={mergeClassName(this.props, 'la-panel', this.state.isExpanded ? '' : 'is-collapsed')}>
                {this.props.children}
            </div>
        </Context.Provider>);
    }
}

Panel.defaultProps = {
    defaultExpanding: true,
    collapsible: true,
    showIcon: true
};

Panel.propTypes = {
    collapsible: PropTypes.bool,
    defaultExpanding: PropTypes.bool,
    showIcon: PropTypes.bool
};

export default Panel;
