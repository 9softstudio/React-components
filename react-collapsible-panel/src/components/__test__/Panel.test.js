import React from 'react';
import Panel from '../Panel';
import Context from '../../panel-context';
import renderer from 'react-test-renderer';

describe('Panel:', () => {
    test('renders correctly', () => {
        const component = renderer.create(<Panel />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('default collapsed', () => {
        const component = renderer.create(<Panel defaultExpanding={false} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('custom class name', () => {
        const component = renderer.create(<Panel className="test" />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('has children', () => {
        const component = renderer.create(<Panel><span>helloworld</span></Panel>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('props to context', () => {
        const component = renderer.create(
            <Panel defaultExpanding={false} collapsible={false}>
                <Context.Consumer>{value => <span contextValue={value} />}</Context.Consumer>
            </Panel>
        );
        expect(component.toJSON().children[0].props.contextValue).toEqual({
            isExpanded: false,
            collapsible: false,
            onToggleCollapse: component.getInstance().onToggleCollapse
        });
    });
});