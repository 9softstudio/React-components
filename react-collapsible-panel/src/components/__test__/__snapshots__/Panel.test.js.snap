// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel: custom class name 1`] = `
<div
  className="test la-panel "
/>
`;

exports[`Panel: default collapsed 1`] = `
<div
  className="la-panel is-collapsed"
/>
`;

exports[`Panel: has children 1`] = `
<div
  className="la-panel "
>
  <span>
    helloworld
  </span>
</div>
`;

exports[`Panel: renders correctly 1`] = `
<div
  className="la-panel "
/>
`;
